cpp/include/logistic_regression.h

```diff
@@ -38,28 +38,32 @@ namespace Nice {
 // Abstract class of common logistic regression functions
 template<typename T>
 class LogisticRegression {
+ private:
+  static Matrix<T> Sigmoid(const Matrix<T> &x){
```

Comment:
Could you explain what the sigmoid function does here? Does it take a whole matrix? In the coursera, it should be taking a scalar. I think you want to instead do an element-wise or row-wise sigmoid here?